torch/jit/frontend.py

```diff
@@ -346,17 +355,19 @@ def build_param_list(ctx, py_args, self_name):
             if arg is not None:
                 ctx_range = build_expr(ctx, arg).range()
                 raise NotSupportedError(ctx_range, _vararg_kwarg_err)
-    result = [build_param(ctx, arg, self_name, False) for arg in py_args.args]
-    result += [build_param(ctx, arg, self_name, True) for arg in py_args.kwonlyargs]
+    result = [build_param(ctx, arg, self_name, kwarg_only=False, args_and_types=args_and_types) for arg in py_args.args]
+    result += [build_param(ctx, arg, self_name, kwarg_only=True, args_and_types=args_and_types) for arg in py_args.kwonlyargs]
     return result
 
 
-def build_param(ctx, py_arg, self_name, kwarg_only):
+def build_param(ctx, py_arg, self_name, kwarg_only, args_and_types=None):
     # NB: In Python3 py_arg is a pair of (str arg, expr? annotation)
     name = py_arg.arg
     r = ctx.make_range(py_arg.lineno, py_arg.col_offset, py_arg.col_offset + len(name))
     if getattr(py_arg, 'annotation', None) is not None:
         annotation_expr = build_expr(ctx, py_arg.annotation)
+    elif args_and_types:
+        annotation_expr = Var(Ident(r, args_and_types[name].pop()))
```

Comment:
As discussed offline, move logic of accessing `args_and_types` to `build_param_list` to avoid coupling `build_param` implementation and its mandatory call orders